web_widget_digital_signature/static/src/js/digital_sign.js

```diff
@@ -0,0 +1,131 @@
+    instance.web.form.widgets.add('signature', 'instance.web.form.FieldSignature');
+        instance.web.form.FieldSignature = instance.web.form.FieldBinaryImage.extend({
+        template: 'FieldSignature',
+        render_value: function() {
+            var self = this;
+            var url;
+            if (this.get('value') && !instance.web.form.is_bin_size(this.get('value'))) {
+                url = 'data:image/png;base64,' + this.get('value');
+            }else if (this.get('value')) {
+                var id = JSON.stringify(this.view.datarecord.id || null);
+                self.digita_dataset = new instance.web.DataSetSearch(self, self.view.model, {}, []);
+                self.digita_dataset.read_slice(['id', self.name], {'domain': [['id', '=', id]]}).then(function(records){
+                    _.each(records,function(record){
+                        if(record[self.name]){
+                            images[self.name] = record[self.name]
+                        }else{
+                            images[self.name] = ""
+                        }
+                    })
+                })
+                var field = this.name;
+                if (this.options.preview_image)
+                    field = this.options.preview_image;
+                url = this.session.url('/web/binary/image', {
+                          model: this.view.dataset.model,
+                          id: id,
+                          field: field,
+                          t: (new Date().getTime()),
+                });
+            }else {
+                images[self.name] = ""
+                url = this.placeholder;
+                self.set('value',images[self.name])
+            }
+            var $img = $(QWeb.render("FieldBinaryImage-img", { widget: this, url: url }));
+            this.$el.find('img').remove();
+            if(this.view.get("actual_mode") !== 'edit' && this.view.get("actual_mode") !== 'create'){
+                this.$el.prepend($img);
+            }else if(this.view.get("actual_mode") == 'edit' ){
+                this.$el.find('> img').remove();
+                this.$el.find('> canvas').remove();
+                if(! this.get('value')){
+                    this.$el.find('> img').remove();
+                    $(this.$el[0]).find(".signature").signature();
+                }else if(this.get('value')){
+                    this.$el.prepend($img);
+                }
+            }else if( this.view.get("actual_mode") == 'create'){
+                images = {}
+                this.$el.find('> img').remove();
+                this.$el.find('> canvas').remove();
+                if(! this.get('value')){
+                    this.$el.find('> img').remove();
+                    $(this.$el[0]).find(".signature").signature();
+                }else if(this.get('value')){
+                    this.$el.prepend($img);
+                }
+            }
+            $(this.$el[0]).find('.clear_sign').click(function(){
```

Comment:
`this.$el` already is a jquery object, and it's only one, so it's enough to say that